plutus-core/plutus-core/src/PlutusCore/Core/Type.hs

```diff
@@ -105,6 +109,26 @@ data Type tyname uni ann
     deriving stock (Show, Functor, Generic)
     deriving anyclass (NFData)
 
+-- | Get recursively all the domains and codomains of a type.
+-- @funTySections (A->B->C) = [A, B, C]@
+-- @funTySections (X) = [X]@
+funTySections :: Type tyname uni a -> NE.NonEmpty (Type tyname uni a)
```

Comment:
Since we have `splitApplication`, it's better to name it `splitFunTys`, and put them both in similar modules (if not the same module) so that they are easier to find. As it stands, if I want to do the same thing this function does, it's difficult to know that it exists.

The function in the GHC API that does this is also called `splitFunTys`, which makes it even more discoverable. If I need to split a function type, this name would be the first thing I look for.

Reply:
Yep, good call. I don't think we have an obvious home for it in `PlutusCore`, but we can certainly name it better.